app/controllers/wikis_controller.rb

```diff
@@ -0,0 +1,67 @@
+class WikisController < ApplicationController
+  include Pundit
+  before_action :authenticate_user!, except: [:show, :index]
+
+  def new
+    @wiki = Wiki.new
+  end
+
+  def index
+    @wikis = policy_scope(Wiki)
+  end
+
+  def create
+    @wiki = current_user.wikis.build(wiki_params)
+
+    if @wiki.save
+      redirect_to @wiki
+    else
+      flash.now[:alert] = "Unable to save wiki. Please try again later"
+      render :new
+    end
+  end
+
+  def show
+    begin
+      @wiki = Wiki.find(params[:id])
+      authorize @wiki, :show?
+    rescue ActiveRecord::RecordNotFound => e
+      redirect_to wikis, flash.now[:error] = "Whoops! Could not find that wiki. \n #{e.message}"
+    end
+  end
+
+  def update
+    @wiki = authorize Wiki.find(params[:id]), :update?
```

Comment:
this is a little confusing - does the `authorize` method return the record?

Either way, I'd recommend separating the finding and the authorizing:

```ruby
@wiki = Wiki.find(params[:id])

authorize @wiki
```